src/data/data.test.js

```diff
@@ -0,0 +1,24 @@
+import PropTypes from "prop-types";
+
+import {getFaction} from "../util/jsonUtils";
+import {unitTemplate} from "./dataTemplates";
+
+const originalConsoleError = console.error;
+
+beforeAll(() => {
+  console.error = message => {
+    throw new Error(message);
+  };
+});
+
+afterAll(() => {
+  console.error = originalConsoleError;
+});
+
+it("Validates json", () => {
+    const rebels = getFaction("rebel");
+    rebels.Commander.forEach(element => {
+        console.log(element);
+        PropTypes.checkPropTypes(unitTemplate, element);
```

Comment:
So the console error override is because proptypes console logs an error and you're trying to catch that?

Reply:
Looks like it's a problem others run into as well: https://github.com/facebook/prop-types/issues/28

Reply:
yeah the proptypes check just prints an error, but I want the test to fail, it doesn't return anything so I can't act on the result :(